api/dataset_api.go

```diff
@@ -0,0 +1,144 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/ONSdigital/go-ns/log"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+// DatasetAPI aggregates a client and URL and other common data for accessing the API
+type DatasetAPI struct {
+	Client    *http.Client
+	URL       string
+	AuthToken string
+}
+
+// NewDatasetAPI creates an DatasetAPI object
+func NewDatasetAPI(client *http.Client, url string, authToken string) *DatasetAPI {
```

Comment:
This constructor isnt needed if its just setting the public vars

Reply:
The vars should not have been public. I have made them private.